### PR TITLE
Some security fixes

### PR DESCRIPTION
**Pull request by nreith**
tensorflow==2.8.0
delete private keys
`jupyter lab clean` to remove staging with vulnerable npm packages from builds

### CODE REVIEW

**Comment by mathematicalmichael** [comment]
Hi @nreith,

Thanks for your contribution! I'm reviewing it now. Optimizations like this are much appreciated and hardly ever prioritized over the regular maintenance / upgrades. 

Can you please elaborate more on the proposal for "each gpu lib in its own small conda environment"?

File: src/Dockerfile.usefulpackages [hunk 1]
```diff
@@ -31,6 +31,18 @@ RUN pip install --no-cache-dir jupyter_contrib_nbextensions \
 #  jupyter nbextension enable codefolding/main
 RUN jupyter labextension install @ijmbarr/jupyterlab_spellchecker
 
+# clean staging folder with vulnerable npm packages from jupyterlab installs (builds) above
+RUN jupyter lab clean && \
+# delete private keys left in packages by other developers, which lead to failed security scans
```
Comment (mathematicalmichael) [inline src/Dockerfile.usefulpackages:36]:
```suggestion
# clean staging folder with vulnerable npm packages from jupyterlab installs (builds) above and
# delete private keys left in packages by other developers, which lead to failed security scans
RUN jupyter lab clean && \
```

File: src/Dockerfile.gpulibs [hunk 2]
```diff
@@ -3,8 +3,9 @@ LABEL maintainer="Christoph Schranz <christoph.schranz@salzburgresearch.at>"
 # Install Tensorflow, check compatibility here:
 # https://www.tensorflow.org/install/source#gpu
 # installation via conda leads to errors in version 4.8.2
+# Pinning tensorflow to 2.8.0, which fixes many, many security vulnerabilities in previous versions
```
Comment (mathematicalmichael) [inline src/Dockerfile.gpulibs:6]:
```suggestion
# (2022-03-23) Pinning tensorflow to 2.8.0, which fixes
#  many security vulnerabilities in previous versions
```

**Review by mathematicalmichael — APPROVED** [review]

**Comment by ChristophSchranz** [comment]
Hi @nreith 
Thank you for your proposals!
Ad tensorflow version: The problem is the compatibility with tensorflow, as the GPU-version requires CUDA and 2.6 is the latest supported version (see https://www.tensorflow.org/install/source#gpu)

Ad separated envs: Would be nice, it still depends on the toolstack the user wishes for his/her project. Some may set up a container per project to separate the toolstacks.

Ad reducing the number of RUN-commands, clean and delete private keys: That would cool. The reason it is separated in multiple RUN statements is because of testing the versions of each installation. Let me know about your suggestions!

**Pull request closed without merging**